File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.go
func construct2DArray(original []int, m int, n int) [][]int {
    var res [][]int
    if m*n != len(original) {
        return res
    }
    for i := 0; i < m; i++ {
        res = append(res, original[i*n : (i+1)*n])
    }
    return res
}

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        vector<vector<int>> res;
        if (m * n != original.size())
            return res;
        for (int i = 0; i < m; i++)
            res.push_back(vector<int>(original.begin() + i * n, original.begin() + (i + 1) * n));
        return res;
    }
};